Bug fixing, debian packaging

pbrandoli opened the pull request:
- Major bug in LiberaAttr.h (wrong removal of high bits)
- Debian packaging scripts
- Other changes mainly due to regeneration from XMI with newer version of POGO

hardion commented:
Thanks Paolo with a little bit of delay. It seems you find a major bug.
I will merge your changes but it is a huge commit. Since we have also change the pogo template, it creates some of conflicts. Also we want to merge into the 'next' branch instead of itech.
Next time it is better to do several commit so I can pick up the relevant one.

/Vincent

hardion commented:
Paolo,

I am curious to know why did you remove some attributes like 

DDDecimationFactor
DDTriggerOffset
MAFLength
MAFDelay

hardion commented:
I had to rebase and split your commit. You can check out the cosylab branch in the MAXIV repo.
I will try to add a maximum of your changes but some of them make reference to your proper environment (/home/paolo or /usr/local).

pbrandoli commented:
Hi Vincent,

Unfortunately there is a big single commit because the changes were done on
a SVN repo before we switched to GIT.

However, the most important bug is still not in this pull request (it was
resolved later): the bug consists in each measurement head opening a MCI
connection. Each time a new MCI connection is open then the previous one is
automatically closed, so only one measuring head is working at any given
time. The resolution consists in sharing a single MCI connection for all
the head. Libera Single Pass has the same problem.

The missing attributes were not present in the branch we pulled when we
started working on this.

Cheers

On 4 May 2015 at 19:41, Hardion Vincent <notifications@github.com> wrote:

> I had to rebase and split your commit. You can check out the cosylab
> branch in the MAXIV repo.
> I will try to add a maximum of your changes but some of them make
> reference to your proper environment (/home/paolo or /usr/local).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MaxIV-KitsControls/Libera-BrilliancePlus/pull/6#issuecomment-98792536>
> .
>

hardion commented:
Effectively it is not easy to merge from svn.
Interesting the problem of the MCI communication. Do you call 'head' A, B, C, D or th rf module?
We did not see yet any trouble with the LiberaSinglePass. What is the symptoms?

Thanks again for your feedback,
Vincent

pbrandoli commented:
The problem is related to the RAF3, RAF4, RAF5, RAF6.

When only one instance of the device server is running to control all the
four boards, then each one tries to open its own MCI connection.
Because the MCI connection is per-process, each new open MCI closes the
previous one.

I don't think that the problem appears if you launch four different
instances of the device server (four processes with different instance
names), but it appears when you launch only one instance and then you
register 4 classes in the Tango DB for the same instance.
I don't think that this week I will have the time to push the changes to
you because I' in the middle of an urgent task, but you can pick them up
from the Solaris repository, or wait a bit until I can do it for you
(hopefully next week).
The solution consists in sharing one MCI for all the RAF boards.

Cheers,
Paolo

On 5 May 2015 at 09:49, Hardion Vincent <notifications@github.com> wrote:

> Effectively it is not easy to merge from svn.
> Interesting the problem of the MCI communication. Do you call 'head' A, B,
> C, D or th rf module?
> We did not see yet any trouble with the LiberaSinglePass. What is the
> symptoms?
>
> Thanks again for your feedback,
> Vincent
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MaxIV-KitsControls/Libera-BrilliancePlus/pull/6#issuecomment-98985845>
> .
>

hardion commented:
I see... This happen when you start one instance exporting several devices.
We don't have such problem because we start one instance per device.

So I will be glad if you include your modification. Create a new issue and a new branch.

/Vincent

hardion commented:
It can wait. 
no problem.

 
> On 05 May 2015, at 09:56, pbrandoli <notifications@github.com> wrote:
> 
> The problem is related to the RAF3, RAF4, RAF5, RAF6.
> 
> When only one instance of the device server is running to control all the
> four boards, then each one tries to open its own MCI connection.
> Because the MCI connection is per-process, each new open MCI closes the
> previous one.
> 
> I don't think that the problem appears if you launch four different
> instances of the device server (four processes with different instance
> names), but it appears when you launch only one instance and then you
> register 4 classes in the Tango DB for the same instance.
> I don't think that this week I will have the time to push the changes to
> you because I' in the middle of an urgent task, but you can pick them up
> from the Solaris repository, or wait a bit until I can do it for you
> (hopefully next week).
> The solution consists in sharing one MCI for all the RAF boards.
> 
> Cheers,
> Paolo
> 
> On 5 May 2015 at 09:49, Hardion Vincent <notifications@github.com> wrote:
> 
> > Effectively it is not easy to merge from svn.
> > Interesting the problem of the MCI communication. Do you call 'head' A, B,
> > C, D or th rf module?
> > We did not see yet any trouble with the LiberaSinglePass. What is the
> > symptoms?
> >
> > Thanks again for your feedback,
> > Vincent
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/MaxIV-KitsControls/Libera-BrilliancePlus/pull/6#issuecomment-98985845>
> > .
> >
> —
> Reply to this email directly or view it on GitHub <https://github.com/MaxIV-KitsControls/Libera-BrilliancePlus/pull/6#issuecomment-98986594>.
>
